pkg/kdump/kdump-view.jsx

```diff
@@ -20,165 +20,230 @@
+                    type: storageLocation,
+                    // HACK: to not needlessly write a path /var/crash as this is the default,
+                    // set an empty string.
+                    path: directory === "/var/crash" ? "" : directory,
+                }
+            },
+            _internal: {
+                ...settings._internal
+            }
         };
-        // we don't support all known storage options currently
-        const storageDest = this.state.storeDest;
-        return (
-            <Form isHorizontal>
+
+        if (storageLocation === "ssh") {
+            newSettings.targets.ssh.server = server;
+            newSettings.targets.ssh.sshkey = sshkey;
+        }
+
+        if (storageLocation === "nfs") {
+            newSettings.targets.nfs.server = server;
+            newSettings.targets.nfs.export = exportPath;
+        }
+
+        handleSave(newSettings)
+                .then(Dialogs.close)
+                .finally(() => setIsSaving(false))
+                .catch(error => {
+                    if (error.details) {
+                        // avoid bad summary like "systemd job RestartUnit ["kdump.service","replace"] failed with result failed"
+                        // if we have a more concrete journal
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error.details))));
+                        error.message = _("Unable to save settings");
+                        error.details = <CodeBlockCode>{ error.details }</CodeBlockCode>;
+                        setError(error);
+                    } else {
+                        // without a journal, show the error as-is
+                        setError(new Error(cockpit.format(_("Unable to save settings: $0"), String(error))));
+                    }
+                });
+    };
+
+    return (
+        <Modal position="top" variant="small" id="kdump-settings-dialog" isOpen
+               title={_("Crash dump location")}
+               onClose={Dialogs.close}
+               footer={
+                   <>
+                       <Button variant="primary"
+                               isLoading={isSaving}
+                               isDisabled={isSaving || !isFormValid || Object.keys(validationErrors).length !== 0}
+                               onClick={saveSettings}>
+                           {_("Save changes")}
+                       </Button>
+                       <Button variant="link"
+                               isDisabled={isSaving}
+                               className="cancel"
+                               onClick={Dialogs.close}>
+                           {_("Cancel")}
+                       </Button>
+                   </>
+               }>
+            {error && <ModalError dialogError={typeof error == 'string' ? error : error.message}
+                                  dialogErrorDetail={typeof error == 'string' ? error : error?.details || null } />}
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18435-20230314-193204-6c24d3a3-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)